cinema/views.py

```diff
@@ -127,20 +133,46 @@ def upload_image(self, request, pk=None):
 
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+    @extend_schema(
+        parameters=[
+            OpenApiParameter(
+                name="title",
+                description="Filter by movie title",
+                type=str,
+            ),
+            OpenApiParameter(
+                name="genres",
+                description="Filter by genres "
+                            "(list of IDs, separated by comma)",
+                type={"type": "array", "items": {"type": "number"}},
+            ),
+            OpenApiParameter(
+                name="actors",
+                description="Filter by actors "
+                            "(list of IDs, separated by comma)",
+                type={"type": "array", "items": {"type": "number"}},
+            ),
+        ],
+        description="Retrieve a list of movies. "
+                    "Filter movies by title, genres and actors.",
+    )
+    def list(self, request, *args, **kwargs):
+        return super().list(request, *args, **kwargs)
+
 
 class MovieSessionViewSet(viewsets.ModelViewSet):
     queryset = (
         MovieSession.objects.all()
-        .select_related("movie", "cinema_hall")
-        .annotate(
+        .select_related("movie", "cinema_hall").annotate(
             tickets_available=(
-                F("cinema_hall__rows") * F("cinema_hall__seats_in_row")
+                F("cinema_hall__rows")
+                * F("cinema_hall__seats_in_row")
                 - Count("tickets")
             )
         )
     )
     serializer_class = MovieSessionSerializer
-    authentication_classes = (TokenAuthentication,)
+
```

Comment:
The `permission_classes` attribute is correctly set to `IsAdminOrIfAuthenticatedReadOnly`, which aligns with the intended access control for this viewset.